torch/_inductor/codegen/cpp.py

```diff
@@ -2291,12 +2299,13 @@ def is_bf16_mem_copy(node: torch.fx.Node):
                         )
                 elif _node.target == "to_dtype" and _node.args[-1] in [torch.bfloat16]:
                     (ops, x, _) = _node.args
-                    from_load = _node.all_input_nodes[-1].target == "load"
-                    to_store = all(usr.target == "store" for usr in _node.users)
+                    # to_bf16 + store_bf16, after legalization, it should be converted to
+                    # to_bf16 + to_bf16 + store_bf16(before condition), rather than to_fp32 + to_bf16 + store_bf16.
+                    to_store_bf16 = all(is_bf16_store(user) for user in _node.users)
                     # The legalization always loads the BF16 tensor as FP32 for computation and converts
                     # back to BF16 after the computation. Hence, there should be no computation w/ BF16.
                     # Therefore, we update the to_dtype by replacing the bf16 dtype with fp32.
-                    if not (from_load or to_store):
+                    if not to_store_bf16:
```

Comment:
Can we fix the problem by changing the original logic to `if not to_store`? We only change the dtype to float only when the consumer nodes are compute nodes.